pkgs/unified_analytics/lib/src/analytics_events.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'analytics.dart';
+import 'enums.dart';
+
+class AnalyticsImpl extends AnalyticsImplNoEvents {
+  /// The [Analytics] implementation that includes the
+  /// events that will be sent by tools using this package
+  ///
+  /// Using pre-defined events with preconfigured parameters
+  /// ensures that events being sent are standardized and
+  /// can be audited by referencing the APIs in this file
+  AnalyticsImpl({
+    required super.tool,
+    required super.homeDirectory,
+    String? flutterChannel,
+    String? flutterVersion,
+    required super.dartVersion,
+    required super.platform,
+    required super.toolsMessageVersion,
+    required super.fs,
+    required super.gaClient,
+    required super.enableAsserts,
+  });
+
+  Future<void> sendMemoryEvent({
```

Comment:
This method also need to be defined in `Analytics`, given that that's the type that clients will need to use when invoking the method.

Reply:
Oh good point, I didn't get around to actually using the class in an example, just put this together to have an idea of what it will look like

Reply:
We'll also want to document (in `Analytics`) which tool this is for so that tool authors don't start using `send` methods that exist for use by other tools.